patches/matrix-js-sdk@31.0.0.patch

```diff
@@ -37,20 +37,104 @@ index 9115598fd769bad15ed23293a68d617d5dbf21a5..95c28e59d8414dbc0837f5c85c98dd96
  }
  function anySignal(signals) {
 diff --git a/lib/sync.js b/lib/sync.js
-index f59265e1089ce832c0be8a23aa19813c2d4738cc..e9b70f2b75a89bb561fc420f03b5106a55f34e09 100644
+index f59265e1089ce832c0be8a23aa19813c2d4738cc..630f58767cbb887620883d075e1b947b9f83b944 100644
 --- a/lib/sync.js
 +++ b/lib/sync.js
-@@ -747,6 +747,13 @@ class SyncApi {
+@@ -719,7 +719,7 @@ class SyncApi {
+    * Invoke me to do /sync calls
+    */
+   async doSync(syncOptions) {
+-    while (this.running) {
++    async function _sync() {
+       const syncToken = this.client.store.getSyncToken();
+       let data;
+       try {
+@@ -729,8 +729,8 @@ class SyncApi {
+         data = await this.currentSyncRequest;
+       } catch (e) {
+         const abort = await this.onSyncError(e);
+-        if (abort) return;
+-        continue;
++        if (abort) return true;
++        return;
+       } finally {
+         this.currentSyncRequest = undefined;
+       }
+@@ -747,6 +747,7 @@ class SyncApi {
          nextSyncToken: data.next_batch,
          catchingUp: this.catchingUp
        };
-+
-+      if (syncToken === data.next_batch) {
-+        // there are no updates and we are just idle--wait before 
-+        // continuing so that idle syncs don't DDoS the server
-+        await new Promise(r => setTimeout(r, 1000));
-+      }
 +
```

Comment:
this was a very old attempt to deal with this before I really understood the nature of this problem